marconi-chain-index/src/Marconi/ChainIndex/Api/JsonRpc/Endpoint/CurrentSyncedBlock.hs

```diff
@@ -42,8 +43,30 @@ data GetCurrentSyncedBlockResult = GetCurrentSyncedBlockResult
   , blockHeaderHash :: Maybe (C.Hash C.BlockHeader)
   , slotNo :: Maybe C.SlotNo
   , nodeTip :: Maybe Tip
+  , blockTimestamp :: Maybe Word64
+  , blockEpochNo :: Maybe C.EpochNo
   }
-  deriving (Eq, Ord, Generic, Show, FromJSON, ToJSON)
+  deriving (Eq, Ord, Show)
+
+instance ToJSON GetCurrentSyncedBlockResult where
+  toJSON (GetCurrentSyncedBlockResult (Just bn) (Just bh) (Just sn) tip (Just ts) (Just en)) =
+    let jsonNodeTip =
+          case tip of
+            Nothing -> []
+            Just tip' -> ["nodeTip" .= toJSON tip']
+     in Aeson.object $
+          [ "blockNo" .= bn
+          , "blockTimestamp" .= ts
+          , "blockHeaderHash" .= bh
+          , "slotNo" .= sn
+          , "epochNo" .= en
+          ]
+            <> jsonNodeTip
+  toJSON (GetCurrentSyncedBlockResult _ _ _ tip _ _) =
+    Aeson.object $
+      case tip of
+        Nothing -> []
+        Just tip' -> ["nodeTip" .= toJSON tip']
```

Comment:
While they may be good in the specific case of the translation from the original type, they don't allow roundtrip. Would you be ok to either refactor them so that it becomes the case (I guess you'll have to generalise what you did for `nodeTip` for the other fields) or throw an error. Or, as we use a custom anyway, put all the fields that must be defined together in a dedicated type.

Reply:
Not sure I fully understand, can you be a bit more specific regarding the roundtrip you're talking about?

I'm not a very big fan of this representation either, but I was wary about refactoring this completely since I wasn't sure why it was defined as a tuple of Maybe's instead of a Maybe of a tuple initially.

Reply:
The idea is that you want to have `fromJSON . toJSON` equals `pure`. Here, if some fields are `Nothing` and others are `Just`, this property doesn't hold.
The main reason for a tuple of maybes was to allow an automatic derivation of `FromJSON` and `ToJSON` but if we drop it, we can definitely move back to a `Maybe` of a tuple.